ledger/ledger-api-client/src/main/scala/com/digitalasset/ledger/client/LedgerClient.scala

```diff
@@ -100,10 +103,28 @@ object LedgerClient {
     for {
       ledgerId <- new LedgerIdentityClient(
         stub(LedgerIdentityServiceGrpc.stub(channel), config.token)
-      )
-        .satisfies(config.ledgerIdRequirement)
+      ).satisfies(config.ledgerIdRequirement)
+        .transformWith {
+          case Success(Right(res)) =>
+            Future.successful(res)
+          case Success(Left(err)) =>
+            Future.failed(new IllegalArgumentException(err))
+          // This means we don't have a token for authorization
+          // such that ledger id verification is not possible.
+          // In this case we try to use the ledgerId we want to verify against which is hopefully provided
+          // via the config.
+          case Failure(e @ GrpcException.UNAUTHENTICATED()) =>
```

Comment:
how do you set the ledger id in the JSON API on startup for this to work out? Currently users do not have to specify one and forcing this is both clunky and backwards compat breaking.

Reply:
@realvictorprm currently the JSON API does not set `optionalLedgerId` on startup so this is not going to work without changing that.

Reply:
Shouldn't this be backward compatible, because prior in the case of unauthenticated the application would have failed? My changes just add an avenue to handle this without crashing.

Reply:
> @realvictorprm currently the JSON API does not set optionalLedgerId on startup so this is not going to work without changing that.

I initially planned to add that change in my PR which removes the token file, however I can already add it in this PR.

Reply:
It works if you keep the package token where we will never hit this. What I’m trying to say is that this does not help you without the package token:
1. In that case you cannot query for the ledger id.
2. You also don’t know it on startup since users do not specify it and I don’t think they should have to (clunky & backwards compat breaking unless you still want to support the case where someone specifies a package token).

Reply:
OK I see the point for the remove-token-file PR. That change would force users to instead of specifying the token file, to specify the ledger id.

Reply:
Geez, I already have locally a non working version where I made the ledger id optional, however that is causing a shit-ton of changes across the whole code base.